examples/balance/app.ddlog

```diff
@@ -0,0 +1,12 @@
+yes (vote_id bigint).
+no  (vote_id bigint).
+
+whatever? (
+    candidate bigint
+).
+
+@weight(1)
+whatever(p) :- yes(_).
```

Comment:
To match the paper more closely, I think this should be an imply factor rather than a unary (also turning the two others into variables).
```
@weight(1)
yes(_) => whatever(p) :- TRUE.
```
In the paper we overloaded the datalog head/body syntax with implication, which later has been disambiguated into a separate syntax.  That gives users precise control of mapping data to the model.  (I know, the `:- TRUE` part looks ugly and shouldn't be mandatory)